src/index.cpp

```diff
@@ -3544,6 +3544,17 @@ Option<bool> Index::search(std::vector<query_tokens_t>& field_query_tokens, cons
 
     process_search_results:
 
+    topster->sort();
+    curated_topster->sort();
+
+    Collection::populate_result_kvs(topster, raw_result_kvs, groups_processed, sort_fields_std);
+    Collection::populate_result_kvs(curated_topster, override_result_kvs, groups_processed, sort_fields_std);
+    std::vector<uint32_t> top_k_result_ids, top_k_curated_result_ids;
+    std::vector<facet> top_k_facets;
+
+    get_top_k_result_ids(raw_result_kvs, top_k_result_ids);
+    get_top_k_result_ids(override_result_kvs, top_k_curated_result_ids);
```

Comment:
Let's move these two calls alongside the respective call to `do_facets()`. This way we will only do this computation when `top_k_facets.size() >  0`.